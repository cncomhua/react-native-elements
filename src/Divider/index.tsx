import { Divider, DividerProps } from './Divider';
import { withTheme } from '../config';

export { Divider };
export type { DividerProps };
/**
 * RNE Divider
 *
 * **Preview**
 *
 * ![URL](URL)
 *
 * [**Documentation**](https://reactnativeelements.com/docs/divider)
 */
export default withTheme(Divider, 'Divider');
