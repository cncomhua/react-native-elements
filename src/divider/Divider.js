import React from 'react';
import { View, ViewPropTypes, StyleSheet } from 'react-native';
import colors from '../config/colors';

let styles = {};

const Divider = ({ style }) => (
  <View style={[styles.container, style && style]} />
);

Divider.propTypes = {
  style: ViewPropTypes.style,
};

styles = StyleSheet.create({
  container: {
    height: 1,
    backgroundColor: colors.grey5,
  },
});

export default Divider;
