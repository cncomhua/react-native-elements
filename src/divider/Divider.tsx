import React from 'react';
import { View, StyleSheet, ViewProps } from 'react-native';
import { withTheme } from '../config';
import theme from '../config/theme';
import { RneFunctionComponent } from '../helpers';

export type DividerProps = ViewProps & {
  style?: object | any[];
  inset?: boolean;
  insetType?: 'left' | 'right' | 'middle';
};

const Divider: RneFunctionComponent<DividerProps> = ({
  style,
  theme,
  inset = false,
  insetType = 'left',
  ...rest
}) => (
  <View
    style={StyleSheet.flatten([
      styles.divider,
      style,
      inset &&
        (insetType === 'left'
          ? styles.leftInset
          : insetType === 'right'
          ? styles.rightInset
          : { ...styles.leftInset, ...styles.rightInset }),
    ])}
    {...rest}
  />
);

const styles = StyleSheet.create({
  divider: {
    backgroundColor: theme?.colors?.divider,
    height: StyleSheet.hairlineWidth,
  },
  leftInset: {
    marginLeft: 72,
  },
  rightInset: {
    marginRight: 72,
  },
});

export { Divider };
export default withTheme(Divider, 'Divider');
