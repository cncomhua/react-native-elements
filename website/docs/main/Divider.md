---


id: divider


title: Divider


slug: /divider

---



import Usage from './usage/Divider/Divider.md'



Dividers are visual separators of content.

Use Divider when you want to make a distinction between sections of content.

Our divider offers adding inset, color, orientation and subHeader to the component using props.

Also receives all [View](https://reactnative.dev/docs/view#props) props.



## Usage


<Usage />

---


## Props

### Divider




## Reference

### Divider

