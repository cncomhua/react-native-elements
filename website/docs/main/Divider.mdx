---
id: divider

title: Divider

slug: /divider
---

import Usage from './usage/Divider/Divider.mdx';

import Props from './props/Divider.mdx';

import { Divider } from 'react-native-elements';

import Tabs from '@theme/Tabs';

import TabItem from '@theme/TabItem';

Dividers are visual separators of content.

Use Divider when you want to make a distinction between sections of content.

Our divider offers adding inset, color, orientation and subHeader to the component using props.

Also receives all [View](https://reactnative.dev/docs/view#props) props.

## Usage

<Usage />

## Props

<Props />
