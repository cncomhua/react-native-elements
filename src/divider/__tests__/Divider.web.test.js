import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../../../samples';
import Divider from '../Divider';

const props = {
  noProps: {
    component: Divider,
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
  style: {
    component: Divider,
    props: { style: { backgroundColor: '#071', height: 5 } },
    enzyme: {
      tests: onlySnapshots,
    },
    styleguidist: {},
  },
};

const samples = {
  props,
};

parseSamples(
  {
    Misc: { sectionComponents: { Divider: { samples } } },
  },
  options
);

describe('Divider Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Divider />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with style', () => {
    const component = shallow(<Divider style={{ backgroundColor: 'blue' }} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
    expect(component.props().style.length).toBe(2);
    expect(component.props().style[1].backgroundColor).toBe('blue');
  });
});
