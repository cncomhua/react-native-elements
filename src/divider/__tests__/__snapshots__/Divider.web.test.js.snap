// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should render with style 1`] = `
<View
  style={
    Object {
      "backgroundColor": "blue",
      "height": 1,
    }
  }
/>
`;

exports[`Divider Component should render without issues 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e1e8ee",
      "height": 1,
    }
  }
/>
`;

exports[`Misc: Divider: Props Misc: Divider: [ prop: noProps ] enzyme-mount: snapshot: Misc: Divider: [ prop: noProps ] enzyme-mount: snapshot 1`] = `
<Divider>
  <View
    style={
      Object {
        "backgroundColor": "#e1e8ee",
        "height": 1,
      }
    }
  >
    <div
      className="rn-alignItems-1oszu61 rn-backgroundColor-lq4czt rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-height-109y4c4 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    />
  </View>
</Divider>
`;

exports[`Misc: Divider: Props Misc: Divider: [ prop: noProps ] enzyme-render: snapshot: Misc: Divider: [ prop: noProps ] enzyme-render: snapshot 1`] = `
<div
  class="rn-alignItems-1oszu61 rn-backgroundColor-lq4czt rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-height-109y4c4 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
/>
`;

exports[`Misc: Divider: Props Misc: Divider: [ prop: noProps ] enzyme-shallow: snapshot: Misc: Divider: [ prop: noProps ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e1e8ee",
      "height": 1,
    }
  }
/>
`;

exports[`Misc: Divider: Props Misc: Divider: [ prop: style ] enzyme-mount: snapshot: Misc: Divider: [ prop: style ] enzyme-mount: snapshot 1`] = `
<Divider
  style={
    Object {
      "backgroundColor": "#071",
      "height": 5,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#071",
        "height": 5,
      }
    }
  >
    <div
      className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-height-4v7adb rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
      style={
        Object {
          "backgroundColor": "rgba(0,119,17,1)",
        }
      }
    />
  </View>
</Divider>
`;

exports[`Misc: Divider: Props Misc: Divider: [ prop: style ] enzyme-render: snapshot: Misc: Divider: [ prop: style ] enzyme-render: snapshot 1`] = `
<div
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-height-4v7adb rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  style="background-color:rgba(0,119,17,1)"
/>
`;

exports[`Misc: Divider: Props Misc: Divider: [ prop: style ] enzyme-shallow: snapshot: Misc: Divider: [ prop: style ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#071",
      "height": 5,
    }
  }
/>
`;
